DylibBinder/DBParameter.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using SwiftReflector;
+
+namespace DylibBinder {
+	internal class DBParameter {
+		public DBParameter (string type, string publicName, string privateName, int index, ParameterOptions parameterOptions)
+		{
+			Type = type;
+			PublicName = publicName;
+			PrivateName = privateName;
+			Index = index;
+			IsVariadic = parameterOptions.HasFlag (ParameterOptions.IsVariadic);
+			HasInstance = parameterOptions.HasFlag (ParameterOptions.HasInstance);
+			IsConstructor = parameterOptions.HasFlag (ParameterOptions.IsConstructor);
+			IsEmptyParameter = parameterOptions.HasFlag (ParameterOptions.IsEmptyParameter);
+		}
+
+		public string Type { get; set; }
+		public string PublicName { get; }
+		public string PrivateName { get; }
+		public bool IsVariadic { get; }
+		public int Index { get; }
+
+		public bool HasInstance { get; }
+		public bool IsConstructor { get; }
+		public bool IsEmptyParameter { get; }
+	}
+
+
+	internal class DBParameterList {
+		public DBParameterList (SwiftBaseFunctionType signature, bool hasInstance, bool isConstructor, int index)
+		{
+			Index = index;
+			int parameterIndex = 0;
+
+			if (hasInstance) {
+				Parameters.Add (new DBParameter ("", "", "self", parameterIndex, ParameterOptions.HasInstance));
+				return;
+			} else if (isConstructor) {
+				Parameters.Add (new DBParameter (".Type", "", "self", parameterIndex, ParameterOptions.IsConstructor));
+				return;
```

Comment:
For this two weird constructors. As a person that does not know the code, this is very mysterious. What about adding some factory methods:
```csharp
DBParameter.CreateConstructor() => new DBParameter (".Type", "", "self", parameterIndex, ParameterOptions.IsConstructor)
```
that will make the code much nicer to understand and read:
```csharp
else if (isConstructor) {
  				Parameters.Add ( DBParameter.CreateConstrutor (parameterIndex));
 				return;
}
```